### PR TITLE
YARN-11444. Improve YARN md documentation format.

### PR DESCRIPTION
**Pull request by slfan1989**
JIRA: YARN-11444. Improve YARN md documentation format.

1. Improve the table format to make the readability better.

When reading the md file locally, I encountered format problems and it was not easy to read. I think we should align the columns

> before modification

Not easy to read, not easy to modify

image

> after modification

image

2. Modify some typo errors.

image

3. Modify the list number to display correctly.
image

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 55s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  1s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  38m  4s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 31s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  58m 19s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 15s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 20s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  19m 21s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 38s |  |  The patch does not generate ASF License warnings.  |
|  |   |  81m 17s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5431/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5431 |
| Optional Tests | dupname asflicense mvnsite codespell detsecrets markdownlint |
| uname | Linux 5e94285e9c7d 4.15.0-200-generic #211-Ubuntu SMP Thu Nov 24 18:16:04 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 04cd33a0b7fcc3ee1d3f4dabc9e5d25eeba67142 |
| Max. process+thread count | 560 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5431/1/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 52s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  41m 35s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 26s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  64m 29s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 13s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 18s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  23m  3s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 36s |  |  The patch does not generate ASF License warnings.  |
|  |   |  91m 13s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5431/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5431 |
| Optional Tests | dupname asflicense mvnsite codespell detsecrets markdownlint |
| uname | Linux e5ba83a0e135 4.15.0-200-generic #211-Ubuntu SMP Thu Nov 24 18:16:04 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 4083fcab2771ba4aabcbdac91f98db3e4acc0eec |
| Max. process+thread count | 595 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5431/2/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by slfan1989** [comment]
@ayushtkn Can you help review this pr? Thank you very much!

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/DevelopYourOwnDevicePlugin.md [hunk 1]
```diff
@@ -139,7 +139,7 @@ public interface DevicePluginScheduler {
 }
 ```
 The above code shows the `DevicePluginScheduler` interface that you might
-needed if you want to arm the plugin with a more efficient scheduler.
+be needed if you want to arm the plugin with a more efficient scheduler.
```
Comment (ayushtkn) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/DevelopYourOwnDevicePlugin.md:142]: does this sound better?
```
The above code shows the you might need  `DevicePluginScheduler` interface  if you want to arm the plugin with a more efficient scheduler.
```

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/TimelineServer.md [hunk 2]
```diff
@@ -59,23 +59,23 @@ Generic data is published by the YARN Resource Manager to the timeline store
...
+8. The single-server implementation of the Timeline Server places a limit on
   the scalability of the service; it also prevents the service being
   High-Availability component of the YARN infrastructure.
 
 Future Plans
 
 1. Future releases will introduce a next generation timeline service
 which is scalable and reliable, ["Timeline Service v2"](./TimelineServiceV2.html).
-1. The expanded features of this service *may not* be available to
+2. The expanded features of this service *may not* be available to
```
Comment (ayushtkn) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/TimelineServer.md:78]: this sounds correct only, the numbering is correct only
http://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/TimelineServer.html

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/TimelineServer.md [hunk 3]
```diff
@@ -833,13 +833,13 @@ Use the following URI to obtain the event objects of the given `entityType`.
...
+2. `limit` - A limit on the number of events to return for each entity. If null,
   defaults to 100 events per entity.
-1. `windowStart` - If not null, retrieves only events later than the given time
+3. `windowStart` - If not null, retrieves only events later than the given time
   (exclusive)
-1. `windowEnd` - If not null, retrieves only events earlier than the given time
+4. `windowEnd` - If not null, retrieves only events earlier than the given time
   (inclusive)
-1. `eventType` - Restricts the events returned to the given types. If null,
+5. `eventType` - Restricts the events returned to the given types. If null,
```
Comment (ayushtkn) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/TimelineServer.md:842]: I doubt if this is required. Pass me the link from here where it shows messed up
http://apache.github.io/hadoop

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/FairScheduler.md [hunk 4]
```diff
@@ -146,7 +146,7 @@ The allocation file must be in XML format. The format contains five types of ele
...
 
-* **A queuePlacementPolicy element**: which contains a list of rule elements that tell the scheduler how to place incoming apps into queues. Rules are applied in the order that they are listed. Rules may take arguments. All rules accept the "create" argument, which indicates whether the rule can create a new queue. "Create" defaults to true; if set to false and the rule would place the app in a queue that is not configured in the allocations file, we continue on to the next rule. The last rule must be one that can never issue a continue. Valid rules are:
+* **A queuePlacementPolicy element**: which contains a list of rule elements that tell the scheduler how to place incoming apps into queues. Rules are applied in the order that they are listed. Rules may take arguments. All rules accept the "create" argument, which indicates whether the rule can create a new queue. "Create" defaults to true; if set to false and the rule would place the app in a queue that is not configured in the allocations file, we continue on to the next rule. The last rule must be one that can never issue a continued. Valid rules are:
```
Comment (ayushtkn) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/FairScheduler.md:149]: continue to continued?Is that a command which is continued but not continue, can you pass me the link to that?

**Review by ayushtkn — CHANGES_REQUESTED** [review]
Couldn't check line by line. It is huge.......

I doubt the numbering stuff, putting 1. is the md files automatically sort the numbering. If not pass me the links on every change from here, where it is unsorted.
http://apache.github.io/hadoop

Spacing changes, I couldn't figure out either, you need to attach screenshots so that folks can figure out what is the impact of having those spaces.

Hope you have built the website locally and checked nothing on those pages is getting broken either

**Comment by slfan1989** [comment]
This pr mainly modifies the format of the table in markdown, so as to better modify the content of the table in md later.

- Take CapacityScheduler.md as an example

image

The main problem is that the tables are not aligned and we need to find the table boundaries. This is sometimes a bit difficult, if the description information is more. We need to find the boundary, when we want to modify some content in the table.

image

If we can align the table, it will be easier to find a row to modify.

image

After modification, we found that the same content is more tidy.

- This pr modifies 27 md files, and the html files are listed in the table

| md file                        | link                                                                                        |
| -------------                  | -------------                                                                               |
| CapacityScheduler.md           | http://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/CapacityScheduler.html          |
| DevelopYourOwnDevicePlugin.md  | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/DevelopYourOwnDevicePlugin.html|
| DockerContainers.md            | http://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/DockerContainers.html           |
| FairScheduler.md               | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/FairScheduler.html             |
| Federation.md                  | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/Federation.html                |
| GracefulDecommission.md        | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/GracefulDecommission.html      |
| NodeLabel.md                   | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/NodeLabel.html                 |
| NodeManager.md                 | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/NodeManager.html               |
| NodeManagerCgroups.md          | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/NodeManagerCgroups.html        |
| NodeManagerRest.md             | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/NodeManagerRest.html           |
| ResourceManagerHA.md           | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/ResourceManagerHA.html         |
| ResourceManagerRest.md         | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/ResourceManagerRest.html       |
| ResourceManagerRestart.md      | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/ResourceManagerRestart.html    |
| ResourceModel.md               | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/ResourceModel.html             |
| RuncContainers.md              | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/RuncContainers.html            |
| SharedCache.md                 | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/SharedCache.html               |
| TimelineServer.md              | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/TimelineServer.html            |
| TimelineServiceV2.md           | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/TimelineServiceV2.html         | 
| UsingFPGA.md                   | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/UsingFPGA.html                 |
| UsingGpus.md                   | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/UsingGpus.html                 |
| UsingNuma.md                   | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/UsingNuma.html                 |
| WebApplicationProxy.md         | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/WebApplicationProxy.html       |
| WebServicesIntro.md            | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/WebServicesIntro.html          |
| WritingYarnApplications.md     | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/WritingYarnApplications.html   |
| YarnApplicationSecurity.md     | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/YarnApplicationSecurity.html   |
| YarnCommands.md                | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/YarnCommands.html              |
| YarnUI2.md                     | https://apache.github.io/hadoop/hadoop-yarn/hadoop-yarn-site/YarnUI2.html                   |

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   1m  2s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  1s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  41m  9s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 29s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  61m 12s |  |  branch has no errors when building and testing our client artifacts.  |
| -0 :warning: |  patch  |  61m 30s |  |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 14s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 19s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  22m 46s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 39s |  |  The patch does not generate ASF License warnings.  |
|  |   |  88m 53s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5431/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5431 |
| Optional Tests | dupname asflicense mvnsite codespell detsecrets markdownlint |
| uname | Linux 40c781589c50 4.15.0-206-generic #217-Ubuntu SMP Fri Feb 3 19:10:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / da6ccfb6c7a01b9c9e44ef69c66141c139bccf3d |
| Max. process+thread count | 626 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5431/3/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request closed without merging**